### PR TITLE
docs: Add additional example/note for Timer's callback usage

### PR DESCRIPTION
**Pull request by nadrimajstor**
Add example: using named function for the Timer's callback.
Add note: improving traceback inside interrupt timers.

### CODE REVIEW

File: docs/library/pyb.Timer.rst [hunk 1]
```diff
@@ -18,6 +18,13 @@ Example usage to toggle an LED at a fixed frequency::
     tim.init(freq=2)                # trigger at 2Hz
     tim.callback(lambda t:pyb.LED(1).toggle())
 
+Example using named function for the callback::
+
+    def tick(timer):                # we will receive the timer object when being called
+        print(timer.counter())      # show current timer's counter value
+    tim = pyb.Timer(4, freq = 1)    # create a timer object using timer 4 - trigger at 1Hz
```
Comment (pfalcon) [inline docs/library/pyb.Timer.rst]: PEP8 requires no spaces around "=" with keyword arguments. And as these is supposed to give example to novices to follow, it's particularly not good idea to use it like that here.

File: docs/library/pyb.Timer.rst [hunk 2]
```diff
@@ -32,6 +39,10 @@ Further examples::
 the servo driver, and Timer 6 is used for timed ADC/DAC reading/writing.
 It is recommended to use the other timers in your programs.
 
+*Note:* Since even the exception handler can't allocate memory during interrupts,
+in order to get good traceback inside the interrupt timers use:
+``import micropython; micropython.alloc_emergency_exception_buf(100)``
```
Comment (pfalcon) [inline docs/library/pyb.Timer.rst]: Again, as this change is targeted at novice users, would make sense to be more detailed where this code should be put. Even I would get an idea that it should be put either inside timer callback, or exception handler altogether.

File: docs/library/pyb.Timer.rst (hunk 1 above)
Comment (nadrimajstor) [inline docs/library/pyb.Timer.rst]: Novice on this side just learned something new. Thank you.

File: docs/library/pyb.Timer.rst (hunk 2 above)
Comment (nadrimajstor) [inline docs/library/pyb.Timer.rst]: Will rephrase to take this (now) obvious issue into account.

**Comment by dpgeorge** [comment]
Fixed up, rebased and merged in e178ef252094b8235295d21a72621a58b92623ab.

**Pull request closed without merging**